fennel/client/client.py

```diff
@@ -844,6 +881,29 @@ def _post(
         check_response(response)
         return response
 
+    def _patch(
+        self,
+        path: str,
+        data: Any,
+        headers: Dict[str, str],
+        compress: bool = False,
+        timeout: float = _DEFAULT_TIMEOUT,
+    ):
+        if compress:
+            data = gzip.compress(data)
+            headers["Content-Encoding"] = "gzip"
+        if self.token:
+            headers["Authorization"] = "Bearer " + self.token
+        response = self.http.request(
+            "PATCH",
```

Comment:
set secrets would be PUT though right

Reply:
set should be through PUT but makes sense to remove set, to maintain consistent APIs.